From 80907affc6d24f3580dd59f75d4203df491846ae Mon Sep 17 00:00:00 2001
From: Simon Richter <Simon.Richter@hogyros.de>
Date: Wed, 19 Jun 2013 11:42:40 +0200
Subject: [PATCH 09/12] Enable cl_khr_fp64 extension for OpenCL stdlib header
To: beignet@lists.freedesktop.org

This allows the stdlib header to define overloads for doubles.
---
 backend/src/ocl_stdlib.h |    4 ++++
 1 file changed, 4 insertions(+)

Index: beignet-0.1+git20130619+42967d2/backend/src/ocl_stdlib.h
===================================================================
--- beignet-0.1+git20130619+42967d2.orig/backend/src/ocl_stdlib.h	2013-06-19 21:04:40.602666635 +0200
+++ beignet-0.1+git20130619+42967d2/backend/src/ocl_stdlib.h	2013-06-19 21:04:48.974666262 +0200
@@ -53,6 +53,8 @@
 #define private __private
 #endif
 
+#pragma OPENCL EXTENSION cl_khr_fp64 : enable
+
 /////////////////////////////////////////////////////////////////////////////
 // OpenCL built-in vector data types
 /////////////////////////////////////////////////////////////////////////////
@@ -4430,6 +4432,8 @@
 DECL_IMAGE(uint4, ui)
 DECL_IMAGE(float4, f)
 
+#pragma OPENCL EXTENSION cl_khr_fp64 : disable
+
 #undef DECL_IMAGE
 #undef DECL_READ_IMAGE
 #undef DECL_READ_IMAGE_NOSAMPLER
